### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by danielrbradley**
- Pull in last 3 months of pulumi/pulumi upgrades
- Upgrade pulumi/java (2 patches only).
- Handle new codegen failures more reliably.

### CODE REVIEW

**Review by thomas11 — APPROVED** [review]
LGTM modulo some small questions/remarks

File: provider/pkg/gen/examples.go [hunk 1]
```diff
@@ -250,7 +251,7 @@ func generateExamplePrograms(example resources.AzureAPIExample, body *model.Body
 
 	program, diags, err := hcl2.BindProgram(parser.Files, bindOptions...)
 	if err != nil {
-		log.Fatalf("failed to bind program: %v", err)
+		return nil, fmt.Errorf("failed to bind program for language for example %s. %v", example.Location, err)
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/gen/examples.go:259]: Oops copy/paste error from the message below

File: provider/pkg/gen/examples.go [hunk 2]
```diff
@@ -236,7 +236,13 @@ type programGenFn func(*hcl2.Program) (map[string][]byte, hcl.Diagnostics, error
 func generateExamplePrograms(example resources.AzureAPIExample, body *model.Body, languages []string,
 	bindOptions ...hcl2.BindOption) (languageToExampleProgram, error) {
 	programBody := fmt.Sprintf("%v", body)
-	debug.Log(programBody)
+	switch example.Location {
+	// TODO: Remove this once https://github.com/pulumi/pulumi/issues/12832 is fixed.
+	case "azure-rest-api-specs/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/examples/metadata/PutMetadata.json":
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/gen/examples.go:241]: Was expecting lots of failures but then turned out to only be one 😄

**Comment by danielrbradley** [comment]
Merged manually as part of https://github.com/pulumi/pulumi-azure-native/commit/f59975b59215770819e3834960dc2fe6477463df

**Pull request closed without merging**